### PR TITLE
Why scala 3 tabs code

### PR DESCRIPTION
**Pull request by benluo**
add tabs for code

### CODE REVIEW

File: _zh-cn/overviews/scala3-book/fp-immutable-values.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+如图所示，您不会改变原始列表 `a`。
+相反，您将过滤和转换函数应用于 `a` 以创建一个新集合，并将该结果分配给新的不可变变量 `b` 。
+
+同样，在 FP 中，您不会创建具有可变 `var` 构造函数参数的类。
+也就是说，你不要这样写：
+
+```scala
+```scala
```
Comment (bishabosha) [inline _zh-cn/overviews/scala3-book/fp-immutable-values.md:38]: duplicate code fence start
```suggestion
```

**Comment by bishabosha** [comment]
Im not so sure about adding translations until the book is stable, which could be another few months from now, what do you think?

**Comment by bishabosha** [comment]
@jxnu-liguobin there are many files changed in this PR that don't match just the "why scala 3" page, is that intentional?

**Comment by jxnu-liguobin** [comment]
> @jxnu-liguobin there are many files changed in this PR that don't match just the "why scala 3" page, is that intentional?

In fact, I don't know why this PR is exists. The new commits seem for other issues.  Because this PR is based on a PR that has not yet been merged.

**Comment by SethTisue** [comment]
needs rebase now that #2394 is merged

**Comment by jxnu-liguobin** [comment]
> needs rebase now that #2394 is merged

Please rebase, then can be reviewd

**Comment by bishabosha** [comment]
I just did the rebase - I see now it really is just adding code tabs, so ill review

File: _overviews/scala3-book/why-scala-3.md [hunk 2]
```diff
@@ -57,41 +60,55 @@ xs.filter(_ < 3)      // List(1, 2)
...
 ```scala
 def add(a: Int, b: Int) = a + b
 ```
+{% endtab %}
+{% endtabs %}
 
 This is more true than ever in Scala 3, such as when using [union types][union-types]:
 
+{% tabs union %}
+{% tab 'Scala 3' for=union %}
```
Comment (bishabosha) [inline _overviews/scala3-book/why-scala-3.md:111]:
```suggestion
{% tab 'Scala 3 Only' for=union %}
```

File: _overviews/scala3-book/why-scala-3.md [hunk 3]
```diff
@@ -103,21 +120,27 @@ def help(id: Username | Password) =
...
 val b = "Hello, world"
 val c = List(1,2,3)
 ```
+{% endtab %}
+{% endtabs %}
 
 Creating types like traits, classes, and enumerations are concise:
 
+{% tabs enum %}
+{% tab 'Scala 3' for=enum %}
```
Comment (bishabosha) [inline _overviews/scala3-book/why-scala-3.md:143]:
```suggestion
{% tab 'Scala 3 Only' for=enum %}
```

File: _overviews/scala3-book/why-scala-3.md [hunk 4]
```diff
@@ -301,25 +339,52 @@ a.filter(_ < 30).map(_ * 10)
...
+{% endtab %}
+{% endtabs %}
 
 `match` expressions let you use pattern matching, and they truly are _expressions_ that return values:
 
+{% tabs match class=tabs-scala-version %}
+{% tab 'Scala 2' for=match %}
+```scala
+val numAsString = i match {
+    case 1 | 3 | 5 | 7 | 9 => "odd"
```
Comment (bishabosha) [inline _overviews/scala3-book/why-scala-3.md:351]: fix indentation of the whole match

File: _overviews/scala3-book/why-scala-3.md [hunk 5]
```diff
@@ -301,25 +339,52 @@ a.filter(_ < 30).map(_ * 10)
...
+{% endtab %}
+{% endtabs %}
 
 Because they can return values, they’re often used as the body of a method:
 
+{% tabs match-body class=tabs-scala-version %}
+{% tab 'Scala 2' for=match-body %}
+```scala
+def isTruthy(a: Matchable) = a match {
+    case 0 | "" => false
```
Comment (bishabosha) [inline _overviews/scala3-book/why-scala-3.md:374]: fix the indentation

File: _overviews/scala3-book/why-scala-3.md [hunk 6]
```diff
@@ -390,6 +452,8 @@ _Safety_ is related to several new and changed features:
 
 Good examples of _ergonomics_ are enumerations and extension methods, which have been added to Scala 3 in a very readable manner:
 
+{% tabs extension %}
+{% tab 'Scala 3' for=extension %}
```
Comment (bishabosha) [inline _overviews/scala3-book/why-scala-3.md:456]:
```suggestion
{% tab 'Scala 3 Only' for=extension %}
```

**Review by bishabosha — APPROVED** [review]
Thanks a lot!

**Pull request merged**